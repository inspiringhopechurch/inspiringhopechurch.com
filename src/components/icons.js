import { library } from "@fortawesome/fontawesome-svg-core";
import {
  faAngleDown,
  faAngleUp,
  faBadgePercent,
  faCommentAltLines,
  faEnvelope,
  faHeart,
  faMapMarkerAlt,
  faUser,
  faPaperPlane,
  faPhone,
  faHeadSideMask,
  faPeopleArrows,
  faPumpSoap,
  faSprayCan,
} from "@fortawesome/pro-solid-svg-icons";
import { faFacebook, faFacebookF, faTwitter, faLinkedinIn, faStripe } from "@fortawesome/free-brands-svg-icons";

// Initialize icon library to use throughout our app
library.add(
  faHeadSideMask,
  faPeopleArrows,
  faPumpSoap,
  faSprayCan,
  faAngleDown,
  faAngleUp,
  faBadgePercent,
  faCommentAltLines,
  faEnvelope,
  faHeart,
  faMapMarkerAlt,
  faUser,
  faPaperPlane,
  faPhone,
  faFacebook,
  faFacebookF,
  faTwitter,
  faLinkedinIn,
  faStripe
);
