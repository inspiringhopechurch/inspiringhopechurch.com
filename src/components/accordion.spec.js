import React from "react";
import { mount } from "enzyme";
import Accordion, { invalidChars } from "./accordion";
// Import fontawesome library files
import "./icons";

describe("Accordion", () => {
  const title = "My Title",
    expanded = false;

  beforeEach(() => {
    //myAccordion = shallow(<Accordion />);
  });

  it("should render without crashing", () => {
    const myAccordion = mount(<Accordion title={title}>Test content</Accordion>);
    expect(myAccordion.find(".accordion.expanded")).toHaveLength(0);
    expect(myAccordion.prop("title")).toBe(title);
    expect(myAccordion.prop("isExpanded")).toBe(expanded);
    expect(myAccordion.find(".accordion svg.fa-plus")).toHaveLength(1);
    expect(myAccordion.find(".accordion svg.fa-minus")).toHaveLength(0);
    expect(myAccordion).toMatchSnapshot();
  });

  it("should have expanded class when isExpanded is true", () => {
    const myAccordion = mount(
      <Accordion title={title} isExpanded={!expanded}>
        Test content
      </Accordion>
    );
    expect(myAccordion.prop("title")).toBe(title);
    expect(myAccordion.prop("isExpanded")).toBe(!expanded);
    expect(myAccordion.find(".accordion.expanded")).toHaveLength(1);
    expect(myAccordion.find(".accordion svg.fa-plus")).toHaveLength(0);
    expect(myAccordion.find(".accordion svg.fa-minus")).toHaveLength(1);
  });
});
