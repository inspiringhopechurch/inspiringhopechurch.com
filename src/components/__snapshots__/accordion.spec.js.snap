// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should render without crashing 1`] = `
<Accordion
  isExpanded={false}
  title="My Title"
>
  <h2
    className="accordion is-size-4 is-uppercase"
  >
    <button
      aria-expanded="false"
      onClick={[Function]}
    >
      My Title
       
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Array [
            "fas",
            "plus",
          ]
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="sm"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus fa-w-14 fa-sm "
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon>
    </button>
  </h2>
  <div
    className="accordion-content"
    data-id="My-Title"
    style={Object {}}
  >
    Test content
  </div>
</Accordion>
`;
